packages/app/src/app/location-selection/selection-criteria/selection-criteria.component.ts

```diff
@@ -169,10 +169,14 @@ export class SelectionCriteriaComponent {
   private reset$ = new Subject<void>();
 
   constructor() {
-    this.form = this.formBuilder.group({
+    // add type safety where payload property names must match.
+    type SharedProps = Pick<SharedCriteria, 'region' | 'reef_type'> | { siteSuitability: any };
+    this.form = this.formBuilder.group<Record<keyof SharedProps, any>>({
```

Comment:
this will now have type error if we were to do something like rename `reef_type`